### ISSUE TITLE
Fault Handler

### ISSUE THREAD

**Issue by Quaker762** [wornwinter/DMGe#22]
I was just thinking, should we include a fault handler for the MMU? Say the (GameBoy) program attempts to write to read-only memory, the fault handler will kick and scream instead of just silently returning, as it's undefined behavior.

**Comment by Quaker762** [comment]
Actually, perhaps the fault handler should be written to accompany ALL types of error, GPU illegal writes, Illegal instruction ETC. This would make error detection more centralised instead of functions all over the place lol.

**Comment by wornwinter** [comment]
Hmm I'm not sure. Maybe I'd have it as part of a debugger later on. A real game boy would just ignore the memory write attempts if it was to ROM. 

By the way, great work on all those new opcodes!

**Comment by Quaker762** [comment]
Yeah that sounds a bit more logical. hahaha thanks I'm making sure not to make too many typos lol.

**Comment by wornwinter** [comment]
Code so far looks great. All seems to work well too :).

On another note, so close to finishing the non CB opcodes lol.

**Comment by Quaker762** [comment]
I know right. Then we have to deal with the CB opcodes.....

Also, is the gameboy palette meant to be the way it is now? At the moment what's meant to be white is green lol. I can fix it by setting the r, g and b values to the same value, but I'm unsure if you want to keep it like this for debugging.
